kaitai_utils.py

```diff
@@ -63,39 +60,69 @@ def get_kaitai_class():
         inspect.getmembers(import_result)))[0][1]
 
 
-def to_json_string(data_binary: BinaryIO, class_type: Type[KaitaiStruct]) -> str:
-    """
-    Parses a binary data object to a JSON string
-
-    @param data_binary: binary data containing the file content
-    @param class_type: class that contains the parsing to a KaiTai struct
-    @return: JSON string representing contents of data object
-    """
-    parsed_kaitai_struct = class_type.from_io(data_binary)
-    return json.dumps(_object_to_dict(parsed_kaitai_struct), indent=2)
-
-
-@streamable_dict
-def _object_to_dict(instance: Any) -> Generator[Dict[str, Any], None, None]:
-    """
-    Recursive helper method that parses an object to a dictionary.
-    Key: The parameters and property method names
-    Value: The parsed value or returning values of the fields and property method names
-
-    @param instance: object that needs parsing to dictionary
-    @return: dictionary containing parsed fields and their respective parsed values in a dictionary
-    """
-    parameters_dict = _parameters_dict(instance)
-    for key, value_object in parameters_dict.items():
-        if not key.startswith("_") and value_object is not None:
-            if _is_kaitai_struct(value_object):
-                yield _to_lower_camel_case(key), _object_to_dict(value_object)
-            elif _is_list(value_object):
-                yield _to_lower_camel_case(key), _list_to_dict(value_object)
-            elif isinstance(value_object, bytes):
-                yield _to_lower_camel_case(key), "data block of size: " + str(len(value_object))
-            else:
-                yield _to_lower_camel_case(key), _process_value(value_object)
+class KaitaiToTraceWriter:
```

Comment:
since this is a private class, perhaps:
```suggestion
class _KaitaiToTraceWriter:
```

Reply:
done